lib/tx_build/claimant.ex

```diff
@@ -0,0 +1,47 @@
+defmodule Stellar.TxBuild.Claimant do
+  @moduledoc """
+  `Claimant` struct definition.
+  """
+  import Stellar.TxBuild.Validations,
+    only: [
+      validate_account_id: 1
+    ]
+
+  alias Stellar.TxBuild.AccountID
+  alias Stellar.TxBuild.ClaimPredicate, as: TxClaimPredicate
+  alias StellarBase.XDR.{ClaimantV0, Claimant, ClaimantType}
+
+  @behaviour Stellar.TxBuild.XDR
+
+  @type validation :: {:ok, any()} | {:error, atom()}
+
+  @type t :: %__MODULE__{destination: String.t(), predicate: TxClaimPredicate.t()}
+
+  defstruct [:destination, :predicate]
+
+  @impl true
+  def new(args, opts \\ [])
+
+  def new({destination, predicate}, _opts),
+    do: new(destination: destination, predicate: predicate)
+
+  def new([destination: destination, predicate: predicate], _opts) do
+    with {:ok, destination} <- validate_account_id({:destination, destination}),
+         {:ok, predicate} <- TxClaimPredicate.validate_predicate(predicate) do
```

Comment:
@juanhurtado10  I'm not sure if delegating the predicate validation to `ClaimPredicate.new/2` is the better solution because in `ClaimPredicate.new/2` we validate if the properties to create a predicate are correct. For this case, we need to validate if a predicate is valid (not its properties), if I add that validation in `ClaimPredicate.new/2` I'm adding a validation that the module `Stellar.TxBuild.ClaimPredicate` does not need.

Being a validation directly associated with `Stellar.TxBuild.Claimant`, I think that the same module would be the best place to add it.

But you know the project better than I do, I don't know if I'm not taking something into account. What do you think?